File: lecture_004(Array)/l_001_basicarray.java
import java.util.Scanner;
public class l_001_basicarray {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        int n = scn.nextInt();
        int[] arr = new int[n];
        input(arr);
        display(arr);
    }
    public static void display(int[] arr) {
        for(int r: arr) {                   //for(int i=0; i<arr.length; i++) {
            System.out.print(r+" ");        //    int r = arr[i];
        }                                   //    System.out.print(r+" ");
        System.out.println();               //}  
    }
    public static void input(int[] arr) {
        for(int i=0; i<arr.length; i++) {
            arr[i] = scn.nextInt();
        }
        display(arr);
    }
}

File: lecture_004(Array)/l_001_basicarray.cpp
#include<iostream>
#include<vector>
using namespace std;
void display(vector<int> &arr) {
    for (int i:arr) {         // for-each loop   //this loop works on value in index
        cout<<i<<" ";         
    }
    cout<<endl;
}
void input(vector<int> &arr) {   //void input(int n);
    for(int i=0; i<arr.size(); i++) {   //this loop work on index
        cin>>arr[i];
    }
    display(arr);
}
int main(int args, char**argv) {
    int n;
    cin>>n;
    vector<int>arr(n,0);    //int n = scn.nextInt();
    input(arr);             //input(n);
    display(arr);           //display(n);

    return 0;
}
